core/packages/contracts/src/TokenVault.sol

```diff
@@ -0,0 +1,41 @@
+// SPDX-License-Identifier: Apache-2.0
+pragma solidity ^0.8.19;
+
+import "openzeppelin/access/Ownable.sol";
+import "openzeppelin/token/ERC20/IERC20.sol";
+import "openzeppelin/token/ERC20/utils/SafeERC20.sol";
+
+/// @title ERC20 Vault
+/// @dev Holds ERC20 Tokens on behalf of ERC20App.
+contract TokenVault is Ownable {
+    using SafeERC20 for IERC20;
+
+    /// @dev Emitted when funds are deposited.
+    event Deposit(address sender, address token, uint256 amount);
+
+    /// @dev Emitted when funds are withdrawn.
+    event Withdraw(address recipient, address token, uint256 amount);
+
+    /// @dev Not enough funds to transfer.
+    error InsufficientBalance();
+
+    /// @dev stores the total balance of each token locked in the vault.
+    mapping(address => uint128) public balance;
+
+    function deposit(address sender, address token, uint128 amount) external onlyOwner {
```

Comment:
Is `onlyOwner` nessessary here for deposit?

Reply:
Its not necessary. I think I will remove it.

Also, am realizing that for consistency with the rest of the code, I should change this contract to use AccessControl instead of Ownable.

Reply:
I think this should be `onlyOwner`. In an upgrade scenario, only one version of the `NativeTokens` contract should own the vault, and such should be the only version allowed to deposit. So if ownership has changed to a v2 of `NativeTokens`, if someone tries to use the v1 contract the `lock` would be rejected.